src/compose/ComposeBox.js

```diff
@@ -217,6 +220,47 @@ class ComposeBox extends PureComponent<Props, State> {
     }
   };
 
+  insertAttachment = async (attachments: DocumentPickerResponse[]) => {
+    this.setState(({ numUploading }) => ({
+      numUploading: numUploading + 1,
+    }));
+
+    const { _, auth } = this.props;
+
+    let placeholders = '';
+    for (let i = 0; i < attachments.length; i++) {
+      const fileName = attachments[i].name ?? `attachment-${i + 1}`;
+      placeholders += `${_('[Uploading {fileName}...]()', { fileName })}\n\n`;
```

Comment:
Oh, but a different translation-related remark:

Let's have the translated string here be just `Uploading {fileName}...`, and interpolate the result inside the `[${…}]()` markup syntax.

That way what the translators have to deal with is simpler -- no extra markup on the outside, which we'd be counting on them to keep verbatim and we don't have a great way to make that clear to them.

And on the other hand that's perfectly fine from the perspective of the question in the https://github.com/zulip/zulip-mobile/pull/4590/files#r622602182 thread, because the extra stuff we're adding is purely syntax from the markup language, not anything that's part of English or the user's natural language.

(Conversely, we do want to keep the `...` inside the translated string, because that's punctuation and it might be expressed differently in another language.)

Reply:
Sure, did these changes.